Library/Homebrew/cmd/commands.rb

```diff
@@ -7,31 +7,52 @@
 module Homebrew
   module_function
 
-  def commands
-    if ARGV.include? "--quiet"
-      cmds = internal_commands + external_commands
-      cmds += internal_developer_commands
-      cmds += HOMEBREW_INTERNAL_COMMAND_ALIASES.keys if ARGV.include? "--include-aliases"
-      puts Formatter.columns(cmds.sort)
-    else
-      # Find commands in Homebrew/cmd
-      puts "Built-in commands"
-      puts Formatter.columns(internal_commands)
-
-      # Find commands in Homebrew/dev-cmd
-      puts
-      puts "Built-in developer commands"
-      puts Formatter.columns(internal_developer_commands)
-
-      # Find commands in the path
-      unless (exts = external_commands).empty?
+  # `@commands_command` stores the code block passed into `define_command` DSL
+  @commands_command = Command.define_command do
+    desc "Show a list of built-in and external commands."
+    option "--quiet" do
+      desc <<-EOS.undent
+        If #{@option} is passed, list only the names of commands without the header.
+      EOS
+      suboption "--include-aliases" do
+        desc <<-EOS.undent
+          With #{@suboption}, the aliases of internal commands will be included.
+        EOS
+      end
+    end
+
+    run do
+      # code which was originally present in Homebrew#commands() method
+      # TODO: generate methods that remove the need for using `ARGV` here
+      if ARGV.include? "--quiet"
```

Comment:
Yes I was thinking the same. Thats why I've included a `TODO` comment. Maybe I can add this in the future PR? Or do you want me to add this in this PR?